### PR TITLE
Split up the azure_deploy workflow

### PR DESCRIPTION
**Pull request by eatyourgreens**
Move builds, deploys and slack notifications into their own reusable workflows. The build job saves the build as a [workflow artifact](https://docs.github.com/en/actions/advanced-guides/storing-workflow-data-as-artifacts), which can be downloaded by the deploy job.

Screenshot of the stages of a staging deploy, showing build, deploy and notification jobs.

Builds are archived by GitHub for 90 days, and can be downloaded from the Actions panel.

Screenshot of the workflow output for a staging build, showing the dist artifact available to download or delete.

Staging branch URL: https://pr-6066.pfe-preview.zooniverse.org

# Required Manual Testing

- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

File: .github/workflows/deploy_branch.yml [hunk 1]
```diff
@@ -5,16 +5,29 @@ on:
     pull_request:
 
 jobs:
-  deploy_staging_branch:
-    name: Deploy staging branch
-    uses: zooniverse/Panoptes-Front-End/.github/workflows/azure_deploy.yml@production-release
+  build:
+    name: Build staging branch
+    uses: zooniverse/Panoptes-Front-End/.github/workflows/npm_build.yml@reusable-workflows
```
Comment (camallen) [inline .github/workflows/deploy_branch.yml:10]: Will the `@reusable-workflows` branch name need to change after this PR is in and the branch cleaned up?

**Review by camallen — APPROVED** [review]
LGTM - minor question on how we use the branch `@reusable-workflows` identifier for re-usable workflows. I assume this will change in a follow up PR once this is in. 

Also I found it a little confusing that the build artifacts are shared via the default value in the source inputs for uploading / downloading artifacts. It may help for understanding the data flow to make these values explicit....

File: .github/workflows/deploy_branch.yml (hunk 1 above)
Comment (eatyourgreens) [reply .github/workflows/deploy_branch.yml:10]: Yeah, once this is deployed the refs will need to change to `production_release`.

**Comment by eatyourgreens** [comment]
Good point about explicitly stating the build directory, which gets reused as the artifact name. I thought it might be annoying to add it to every build, but it is only a couple of lines and it does make the workflows easier to read.

File: .github/workflows/deploy_branch.yml (hunk 1 above)
Comment (eatyourgreens) [reply .github/workflows/deploy_branch.yml:10]: Leaving this here as a reminder to myself to change that ref before I delete this branch.

**Pull request merged**